os/kernel/task/task_prctl.c

```diff
@@ -237,6 +241,32 @@ int prctl(int option, ...)
 #endif
 	}
 	break;
+#ifdef CONFIG_TASK_MONITOR
+	case PR_MONITOR_REGISTER:
+	{
+		task_monitor_update_list(getpid(), MONITORED);
+	}
+	break;
+	case PR_MONITOR_UPDATE:
+	{
+		struct tcb_s *tcb;
+		tcb = sched_gettcb(getpid());
+		if (tcb == NULL) {
+			sdbg("Fail to update the status in task monitor.\n");
+			goto errout;
+		}
+		tcb->is_active = true;
+	}
+	break;
+#else
+	case PR_MONITOR_REGISTER:
```

Comment:
if config wrapped these constant itself, you don't have to make #else part...because caller can't use these constant.
It will go to default directly.